### PR TITLE
one-d-two-d check refactor

### PR DESCRIPTION
**Pull request by wvangerwen**
Updated to be compatible with environment in MI 3.34.5 and 3DI toolbox 2.5.5:
- [x] conda solvable environment
- [x] Fix issues in pytests

### CODE REVIEW

**Comment by d2hydro** [comment]
@wvangerwen; Ik kan de gebroken test technisch fixen (extra filter op numpy array), maar weet niet of dat functioneel gewenst is. Efficiëntste daar samen even naar te kijken Bij de 1e window krijg je `nodelevel` met allemaal `None` in:

```
level[in_interpol_and_suitable] = np.sum(weight * nodelevel, axis=1)
```

Ik heb reproductie in dit bestandje staan: https://github.com/threedi/hhnk-threedi-tools/blob/one_d_two_d-refactor/tests/one_d_two_d_exception.py

File: envs/environment_test.yml
Comment (wvangerwen) [inline envs/environment_test.yml:1]: @DanielTollenaar zijn deze wijzigingen nog terecht?

**Review by wvangerwen — COMMENTED** [review]
@DanielTollenaar kan je deze nog even doornemen?

**Comment by DanielTollenaar** [comment]
Updated env results in following pytest Exceptions:
image

**Comment by DanielTollenaar** [comment]
**ValueError: cannot reindex on an axis with duplicate labels** is raised by:
```
all_manholes.update(intersections.set_index(node_id_col)[type_col])
```

`intersections.set_index(node_id_col)[type_col]` does not have an unique index:
image

`intersections.drop_duplicates(subset=[node_id_col, type_col]).set_index(node_id_col)[type_col]` does:
image

**Comment by DanielTollenaar** [comment]
@wvangerwen, could you have a look at test_run_depth_at_timesteps. This is returning an array with None:

see: [delaunay](https://github.com/threedi/hhnk-threedi-tools/blob/01af596f5be07c0d02ef559db66bc40fe79d1895/hhnk_threedi_tools/core/result_rasters/calculate_raster.py#L61C2-L77C34):

    def delaunay(self):
        """
        Return a (delaunay, s1) tuple.


        `delaunay` is a qhull.Delaunay object, and `s1` is an array of
        waterlevels for the corresponding delaunay vertices.
        """
        try:
            return self.cache[self.DELAUNAY]
        except KeyError:
            points_grid = np.array(self.grid_gdf.centroid.apply(lambda x: [x.x, x.y]).to_list())


            # reorder a la lizard
            points_grid, wlvl = morton.reorder(points_grid, self.wlvl_raw)
            delaunay = qhull.Delaunay(points_grid)
            self.cache[self.DELAUNAY] = delaunay, wlvl
            return delaunay, wlvl

This causes in `block_calculate_delauney_wlvl` an Exception as `nodelevel` as an array with None:
        delaunay, wlvl = self.delaunay
        simplices = delaunay.find_simplex(points_mesh)


        # determine which points will use interpolation
        in_gridcell = nodeid_arr != 0
        in_triangle = simplices != -1
        in_interpol = in_gridcell & in_triangle
        points_int = points_mesh[in_interpol]


        # get the nodes and the transform for the corresponding triangles
        transform = delaunay.transform[simplices[in_interpol]]
        vertices = delaunay.vertices[simplices[in_interpol]]


        # calculate weight, see print(spatial.Delaunay.transform.__doc__) and
        # Wikipedia about barycentric coordinates
        weight = np.empty(vertices.shape)
        weight[:, :2] = np.sum(transform[:, :2] * (points_int - transform[:, 2])[:, np.newaxis], 2)
        weight[:, 2] = 1 - weight[:, 0] - weight[:, 1]


        # set weight to zero when for inactive nodes
        nodelevel = wlvl[vertices]
        weight[nodelevel == NO_DATA_VALUE] = 0


        # determine the sum of weights per result cell
        weight_sum = weight.sum(axis=1)


        # further subselect points suitable for interpolation
        suitable = weight_sum > 0.5
        weight = weight[suitable] / weight_sum[suitable][:, np.newaxis]
        nodelevel = nodelevel[suitable]


        # combine weight and nodelevel into result
        in_interpol_and_suitable = in_interpol.copy()
        in_interpol_and_suitable[in_interpol] &= suitable
        level[in_interpol_and_suitable] = np.sum(weight * nodelevel, axis=1)

image

**Review by DanielTollenaar — APPROVED** [review]
@wvangerwen, if you can fix the delaunay issue as mentioned in the comment, I think we are good to go: https://github.com/threedi/hhnk-threedi-tools/pull/88#issuecomment-2081559204

**Comment by wvangerwen** [comment]
Solved the import warnings and deprecation warnings with simplices and delauney import.

Cannot replicate error in https://github.com/threedi/hhnk-threedi-tools/pull/88#issuecomment-2081559204. Ignoring for now, we need to revisit this part of the code towards the next release.

**Pull request merged**